### PR TITLE
[tt-train] Add kahan summation in AdamW

### PR DESCRIPTION
**Pull request by rfurko-tt**
### Problem description
bfloat16 has wide range, but is not as precise as float32. Some of the parameters are not updated due to small magnitude of the gradients (after multiplication by learning rate), as an example: gamma in LayerNorm. 

### What's changed
Add kahan summation flag to enable kahan summation when update weights. 

### Checklist
- [ ] Post commit CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/12128966379
- [x] New/Existing tests provide coverage for changes

image

### CODE REVIEW

File: tt-train/sources/ttml/optimizers/adamw.cpp [hunk 1]
```diff
@@ -16,13 +16,18 @@ namespace {
...
+const std::string kKahanCompensation = "kahan_compensation/";
 
 }  // namespace
 
 namespace ttml::optimizers {
 
 MorehAdamW::MorehAdamW(autograd::NamedParameters parameters, const AdamWConfig& config) :
     OptimizerBase(std::move(parameters)), m_config(config) {
+    if (m_config.use_kahan_summation) {
+        throw std::runtime_error("MorehAdamW: Kahan summation is not supported.");
```
Comment (dmakoviichuk-tt) [inline tt-train/sources/ttml/optimizers/adamw.cpp:28]: Please add that it is supported with our Composite Adam.

**Review by dmakoviichuk-tt — APPROVED** [review]
